src/Sulu/Bundle/MediaBundle/Media/Storage/StorageInterface.php

```diff
@@ -14,6 +14,12 @@
 /**
  * Defines the operations of the StorageLayer.
  * The StorageLayer is a interface to centralized management of media store.
+ *
+ *  @phpstan-type StorageOptions array{
+ *     "directory" ?: string,
+ *     "segment" ?: int|string,
+ *     "fileName" ?: string,
+ * }
```

Comment:
I'm not sure to add this to the `interface` basically the storageinterface don't care what options to storage use and can be anything for custom storages. So would keep this currently on the FlysystemStorage.

Reply:
I think the best way would be to have an object that could be extended and not just a loose array, but for now your suggestion makes the most sense.

Reply:
object would same problem as the typed array. So if we want not force how somebody store its media and what storageOptions they use we should as said keep the `array<string, string|null>`, this also avoids a bc break in the interface.

If I personally would rework the interface I would even more go with a simple `path` `string` instead of any object or array.

Reply:
Yeah I mean the Flysystem adapter also has a `PathNormalizer` which the user could customize if they wanted some fancy behavior.

Reply:
But it's reverted now.